### PR TITLE
194 frontend post dashboard component

### PR DESCRIPTION
**Pull request by taylorfinelli**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

- `RotatingBarbellIcon` now has `color?`
  - default to `#CACACA`
  - Prettier had its way with this file
- `PostDashboard` component shows snippet of 3 most recent posts from friends
  - Also wrote unit tests (contained in 2 files since I couldn't figure out how to mock same API with different values)
  - Help needed
  - Need one more case to cover, can't figure out
  - Still like 89% coverage tho
- `Label` component used for dashboard components
  - Unit tests
- `getRecentPostsByUserIdAndPostCount` API gets most recent posts from friends
- Changed post model so that it includes a date of creation

# Tests

How was this tested?

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# Screenshots
**Put it on the notifs page for testing**

https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/8f2ce018-dcd7-481f-8a38-1f840d7710ba

image

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# AI Usage
None

### CODE REVIEW

**Comment by thienguen** [comment]
image

we have 2 loaders?

**Comment by taylorfinelli** [comment]
> image
> 
> we have 2 loaders?

Remember, it is being demonstrated on the Notifications page. The component is meant for the Dashboard page. The Dashboard page doesn't have the loader as far as I know

**Comment by thienguen** [comment]
> > image
> > we have 2 loaders?
> 
> Remember, it is being demonstrated on the Notifications page. The component is meant for the Dashboard page. The Dashboard page doesn't have the loader as far as I know


that's one thing too, dashboard will have multiple "cards" does each of them have a loader of their own? I don't think that's good but it can be change when we integrate all "cards" into the dashboard

**Comment by taylorfinelli** [comment]
> > > image
> > > we have 2 loaders?
> > 
> > 
> > Remember, it is being demonstrated on the Notifications page. The component is meant for the Dashboard page. The Dashboard page doesn't have the loader as far as I know
> 
> that's one thing too, dashboard will have multiple "cards" does each of them have a loader of their own? I don't think that's good but it can be change when we integrate all "cards" into the dashboard

Yeah I get you, but I'm not sure what else to put in there while it's loading.

**Comment by thienguen** [comment]
> > > > image
> > > > we have 2 loaders?
> > > 
> > > 
> > > Remember, it is being demonstrated on the Notifications page. The component is meant for the Dashboard page. The Dashboard page doesn't have the loader as far as I know
> > 
> > 
> > that's one thing too, dashboard will have multiple "cards" does each of them have a loader of their own? I don't think that's good but it can be change when we integrate all "cards" into the dashboard
> 
> Yeah I get you, but I'm not sure what else to put in there while it's loading.

I think there will be a common loader till all data in the dashboard is resolved, for now that is fine, or we can simply display `null`

**Review by mva919 — APPROVED** [review]
Thanks for including unit tests

**Comment by thienguen** [comment]
reviewing now

**Comment by taylorfinelli** [comment]
> reviewing now

thanks

**Comment by thienguen** [comment]
image

look realy nice

File: apps/expo/src/components/post/PostDashboard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+jest.mock('~/context/global-context', () => ({
+  useGlobalContext: jest.fn(() => ({
+    userData: { id: 1 },
+  })),
+}))
+
+const mockApi = api as { api: any }
+
+jest.mock('~/utils/trpc/api', () => ({
```
Comment (thienguen) [inline apps/expo/src/components/post/PostDashboard.test.tsx:14]: thank you for adding the changes

File: apps/expo/src/components/ui/label/Label.test.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
+import { render, screen } from '@testing-library/react-native'
+
+import Label from './Label'
+
```
Comment (thienguen) [inline apps/expo/src/components/ui/label/Label.test.tsx:6]: this test is perfect

**Review by thienguen — APPROVED** [review]

**Pull request merged**